### PR TITLE
One, verifyable, DI framework

### PR DESCRIPTION
**Pull request by ChadNedzlek**
Most of our code uses Microsoft.Extensions.DependencyInjection, which is a fairly simplisitic
dependency injection framework. It's also relatively easy to verify, which is good.

There was a smattering of Autofac in a few places, remove those and just normalize on the
one that was used for a majority of things.

This increases substantially our ability to validate registrations to prevent deployment errors.

### CODE REVIEW

File: src/Maestro/DependencyUpdater/DependencyUpdater.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public sealed class DependencyUpdater : IServiceImplementation, IDependencyUpdat
             ILogger<DependencyUpdater> logger,
             BuildAssetRegistryContext context,
             IRemoteFactory factory,
-            Func<ActorId, ISubscriptionActor> subscriptionActorFactory)
+            IActorLookup<ISubscriptionActor> subscriptionActorFactory)
```
Comment (alexperovich) [inline src/Maestro/DependencyUpdater/DependencyUpdater.cs:44]: >IActorLookup [](start = 12, length = 12)

This name is weird. It is much more a factory. It's not looking up actors, it is constructing a proxy to talk to a specific actor.

File: src/Maestro/SubscriptionActorService/PullRequestActor.cs [hunk 2]
```diff
@@ -56,8 +58,15 @@ public Task ReceiveReminderAsync(string reminderName, byte[] state, TimeSpan due
     ///     A service fabric actor implementation that is responsible for creating and updating pull requests for dependency
     ///     updates.
     /// </summary>
-    public class PullRequestActor : IPullRequestActor, IRemindable, IActionTracker
+    public class PullRequestActor : IPullRequestActor, IRemindable, IActionTracker, IStatefulActor
```
Comment (alexperovich) [inline src/Maestro/SubscriptionActorService/PullRequestActor.cs:61]: >IStatefulActor [](start = 84, length = 14)

NIT: the Stateful in this interface is redundant. All Actors are stateful by definition.

File: src/Microsoft.DotNet.ServiceFabric.ServiceHost/DelegatedStatefulService.cs [hunk 3]
```diff
@@ -73,7 +65,7 @@ protected override IEnumerable<ServiceReplicaListener> CreateServiceReplicaListe
             }
 
             Type[] interfaces = typeof(TServiceImplementation).GetAllInterfaces()
-                .Where(iface => iface.IsAssignableTo<IService>())
```
Comment (alexperovich) [inline src/Microsoft.DotNet.ServiceFabric.ServiceHost/DelegatedStatefulService.cs:76]: >IsAssignableTo [](start = 38, length = 14)

I always wondered where this very nice IsAssignableTo method was coming from. I guess autofac was giving us this goodness.

File: src/Microsoft.DotNet.ServiceFabric.ServiceHost/Actors/DelegatedActor.cs [hunk 4]
```diff
@@ -148,42 +136,13 @@ public override void Intercept(IInvocation invocation)
             base.Intercept(invocation);
         }
 
-        protected override async Task InterceptAsync(IInvocation invocation, Func<Task> call)
```
Comment (alexperovich) [inline src/Microsoft.DotNet.ServiceFabric.ServiceHost/Actors/DelegatedActor.cs:151]: Why did you remove these 2 Intercept implementations? This means that not all methods will be intercepted correctly.

Comment (alexperovich) [reply src/Microsoft.DotNet.ServiceFabric.ServiceHost/Actors/DelegatedActor.cs:151]: Ahh, I see you made AsyncInterceptor handle this. Thats okay I guess. It makes it slightly less performant.

---
In reply to: [406538414](https://github.com/dotnet/arcade-services/pull/1099#discussion_r406538414) [](ancestors = 406538414)

Comment (ChadNedzlek) [reply src/Microsoft.DotNet.ServiceFabric.ServiceHost/Actors/DelegatedActor.cs:151]: Yeah, I kept screwing up some of the implementations (first I forgot to resister stuff, then I forgot to call my new method).

I figured this thing is already doing reflectiony-interceptor magic, a wrapped task or two was worth the simplification.

File: src/Microsoft.DotNet.ServiceFabric.ServiceHost/DelegatedStatefulService.cs (hunk 3 above)
Comment (ChadNedzlek) [reply src/Microsoft.DotNet.ServiceFabric.ServiceHost/DelegatedStatefulService.cs:76]: Yeah, I was surprised it vanished too.

File: src/Maestro/SubscriptionActorService/PullRequestActor.cs (hunk 2 above)
Comment (ChadNedzlek) [reply src/Maestro/SubscriptionActorService/PullRequestActor.cs:61]: What's a better name to explain what this type is... IDelegatedActor, IProxiedActor... I'm not 100% what the use case of the intercepty guy is to pick a good word here.

File: src/Maestro/DependencyUpdater/DependencyUpdater.cs (hunk 1 above)
Comment (ChadNedzlek) [reply src/Maestro/DependencyUpdater/DependencyUpdater.cs:44]: K, I'll rename it. :-)

File: src/Maestro/SubscriptionActorService/PullRequestActor.cs (hunk 2 above)
Comment (alexperovich) [reply src/Maestro/SubscriptionActorService/PullRequestActor.cs:61]: The interceptor stuff is so we can log every actor method invocation to app insights, and so we can create a new scope and implementation type per method call. Unfortunately IActor is already used. I can't really think of any other name. Maybe IActorImplementation, thats kinda what the whole type is.

Comment (ChadNedzlek) [reply src/Maestro/SubscriptionActorService/PullRequestActor.cs:61]: Hrm... Uh... ITracedActor? Since it's used to enable the automatic tracing of stuff.

Comment (ChadNedzlek) [reply src/Maestro/SubscriptionActorService/PullRequestActor.cs]: Went with IActorImplementation.  I really wanted to just pass the whole actor, since it's the source of all 3 values, but that makes testing hard, since they don't have an actual actor, and I didn't want to introduce yet another interface IDelegatedActor, for them to construct and pass.
